src/kalendar/pentad.py

```diff
@@ -0,0 +1,217 @@
+    Uses standard yearly definition of pentads where the 365 day
+    calendar year is divided into 73 pentads of 5 days each.
+    Under the hood, each ``Pentad``
+    is stored as the first day of the pentad, the 1st pentad of the year
+    is on January 1st, the 2nd on January 6th, and so on. Leap years have
+    366 days, and pentad 12 is extended comprise 6 days that
+    includes February 29th. Extension of ``datetime.date``
+    allows for comparison of pentads and conversion between
+    classes and use of existing constructors, methods, and properties,
+    with some additions detailed below.
+
+    Base constructor takes year and pentad. Can
+    also be constructed directly from ``datetime.date``
+    or ``datetime.datetime`` objects and use any
+    ``datetime.date`` constructors:
+
+    ``fromdate()``
+    ``fromdatetime()``
+    ``fromisoformat()``
+    ``fromisocalendar()``
+    ``fromtimestamp()``
+    ``fromordinal()``
+
+    Method to return ``datetime.date`` object
+    added, ``todate()``. Other methods remain
+    available. Read-only property ``pentad``
+    added alongside year, month, and day.
+
+    Parameters
+    ----------
+    year : int
+        Year.
+    pentad : int
+        Pentad, from 1 to 73.
+
+    Examples
+    --------
+    >>> p = Pentad(2022, 1)
+    >>> p
+    Pentad(2022, 1)
+    >>> p.todate()
+    datetime.date(2022, 1, 1)
+    >>> p - 1
+    Pentad(2021, 73)
+    >>> p.pentad
+    1
+    """
+
+    def __new__(cls, year: int, pentad: int):
+        """Initialize new ``Pentad``."""
+        if pentad < 1 or pentad > 73:
+            raise ValueError("Pentad must be between 1 and 73.")
+
+        # get pentad month and day from non-leap year
+        dt = datetime.date(year=2010, month=1, day=1) + datetime.timedelta(
+            (pentad - 1) * 5
+        )
+
+        return super().__new__(cls, year, dt.month, dt.day)  # noqa: FKA01
+
+    def __repr__(self):
+        """Represent ``Pentad``."""
+        return f"{self.__class__.__name__}({self.year}, {self.pentad})"
+
+    def __reduce__(self):
+        """Reduce."""
+        return (self.__class__, (self.year, self.pentad))
+
+    def __str__(self):
+        """Print representation."""
+        return f"{self.year} P{self.pentad}"
+
+    def __add__(  # type: ignore[override]
+        self, other: Union[int, datetime.timedelta]
+    ) -> Union[Pentad, datetime.timedelta]:
+        """Addition magic method for pentads.
+
+        For integers, adds that to the pentads and returns a new
+        ``Pentad`` object. Otherwise, follows the behavior of
+        ``datetime.date``.
+
+        Examples
+        --------
+        >>> p = Pentad(2021, 73)
+        >>> p + 1
+        Pentad(2022, 1)
+
+        Returns
+        -------
+        Union[Pentad, datetime.timedelta]
+            Returns ``Pentad`` if adding integer, or
+            ``datetime.timedelta`` otherwise.
+        """
+        if isinstance(other, int):
```

Comment:
Here I would use try / except instead of checking instance, same with sub, and also for `Dekad`